### PR TITLE
GCP ML services integration.

### PR DESCRIPTION
**Pull request by kornelregius**
Add GCP integrations with the latest client major versions of

* Cloud Translation V2 (basic) and V3 (advanced)
* Natural language V2
* Vision V2
* Video intelligence V2

ticket: b/175030896

**Do not merge until corresponding GCP integration controller version released!**

### CODE REVIEW

File: patches/sitecustomize.py [hunk 1]
```diff
@@ -7,7 +7,20 @@
...
+        'google.cloud.translate',
+        'google.cloud.translate_v2',
+        'google.cloud.translate_v3',
+        'google.cloud.language',
+        'google.cloud.language_v1',
+        'google.cloud.videointelligence',
+        'google.cloud.videointelligence_v1',
+        'google.cloud.vision',
+        'google.cloud.vision_v1',
+        ]
```
Comment (Philmod) [inline patches/sitecustomize.py:23]: nit: I _think_ you have to align the `]` with `_MODULES`, or put it after / on the same line as the last element

**Review by Philmod — APPROVED** [review]
LGTM, but let @rosbo reviews it too

File: patches/sitecustomize.py (hunk 1 above)
Comment (kornelregius) [reply patches/sitecustomize.py]: Fixed in ef87f07

File: patches/kaggle_gcp.py [hunk 2]
```diff
@@ -251,10 +287,91 @@ def init_automl():
...
+    if not is_user_secrets_token_set():
+        return vision
+
+    from kaggle_gcp import get_integrations
+    if not get_integrations().has_vision():
+        return vision
+
+    from kaggle_secrets import GcpTarget
+    kernel_credentials = KaggleKernelCredentials(target=GcpTarget.VISION)
+    monkeypatch_client(vision.ImageAnnotatorClient,kernel_credentials)
```
[the remark this answers is not in the document]
Comment (kornelregius) [reply patches/kaggle_gcp.py]: Fixed in ef87f07

File: patches/kaggle_gcp.py [hunk 3]
```diff
@@ -251,10 +287,91 @@ def init_automl():
...
+    from google.cloud import translate_v3
+    if not is_user_secrets_token_set():
+        return translate_v3
+
+    from kaggle_gcp import get_integrations
+    if not get_integrations().has_translation():
+        return translate_v3
+    from kaggle_secrets import GcpTarget
+    kernel_credentials = KaggleKernelCredentials(target=GcpTarget.TRANSLATION)
+    monkeypatch_client(translate_v3.TranslationServiceClient,kernel_credentials)
```
[the remark this answers is not in the document]
Comment (kornelregius) [reply patches/kaggle_gcp.py]: Fixed in ef87f07

File: patches/kaggle_gcp.py [hunk 4]
```diff
@@ -251,10 +287,91 @@ def init_automl():
...
+    from google.cloud import translate_v2
+    if not is_user_secrets_token_set():
+        return translate_v2
+
+    from kaggle_gcp import get_integrations
+    if not get_integrations().has_translation():
+        return translate_v2
+    from kaggle_secrets import GcpTarget
+    kernel_credentials = KaggleKernelCredentials(target=GcpTarget.TRANSLATION)
+    monkeypatch_client(translate_v2.Client,kernel_credentials)
```
[the remark this answers is not in the document]
Comment (kornelregius) [reply patches/kaggle_gcp.py]: Fixed in ef87f07

**Comment by rosbo** [comment]
LGTM. The tests are failing for an orthogonal reason. I will fix master today and let you know when you can rebase your branch with the latest changes. After that, if all the build for your branch is green, you will be able to merge this.

Thank you

**Comment by rosbo** [comment]
@kornelregius I merged #929 which fixes the `master` build.

**Comment by kornelregius** [comment]
It's green now:
https://ci.kaggle.net/blue/organizations/jenkins/kernels%2Fdocker-python/detail/PR-925/7/pipeline

**Pull request merged**